test-site/src/pages/LocationsPage.tsx

```diff
@@ -0,0 +1,76 @@
+import { useSearchActions, Matcher, useSearchState, SelectableStaticFilter } from '@yext/search-headless-react';
+import {
+  AppliedFilters,
+  ResultsCount,
+  SearchBar,
+  StandardCard,
+  VerticalResults,
+  LocationBias,
+  MapboxMap,
+  Pagination,
+  MapboxMapProps,
+} from '@yext/search-ui-react';
+import { LngLat, LngLatBounds } from 'mapbox-gl';
+import { useCallback, useLayoutEffect } from 'react';
+import { MapPin } from '../components/MapPin';
+
+export interface Location {
+  yextDisplayCoordinate: {
+    latitude: number,
+    longitude: number
+  } | undefined
+}
+
+const mapboxOptions: MapboxMapProps<Location>['mapboxOptions'] = {
+  zoom: 10
+};
+
+export function LocationsPage() {
+  const searchActions = useSearchActions();
+  const filters = useSearchState(state => state.filters.static);
+  useLayoutEffect(() => {
+    searchActions.setVertical('KM');
+    searchActions.executeVerticalQuery();
+  }, [searchActions]);
+
+  const onDrag: MapboxMapProps<Location>['onDrag'] = useCallback(
+    (center: LngLat, bounds: LngLatBounds) => {
+      const radius = center.distanceTo(bounds.getNorthEast());
+      const nonLocationFilters: SelectableStaticFilter[] = filters?.filter(f => f.filter.kind !== 'fieldValue' || f.filter.fieldId !== 'builtin.location') ?? [];
+      const nearFilter: SelectableStaticFilter = {
+        selected: true,
+        filter: {
+          kind: 'fieldValue',
+          fieldId: 'builtin.location',
+          matcher: Matcher.Near,
+          value: { ...center, radius }
+        }
+      };
+      searchActions.setStaticFilters([...nonLocationFilters, nearFilter]);
+      searchActions.executeVerticalQuery();
+    }, [filters, searchActions]);
+  return (
+    <div>
+      <SearchBar />
+      <div className='flex flex-col'>
+        <div className='flex items-baseline'>
+          <ResultsCount />
+          <AppliedFilters />
+        </div>
+        <div className='h-80 mb-4'>
+          <MapboxMap<Location>
```

Comment:
do we need to specify the type param or can it be inferred from the props?

Reply:
It helps when I had the props inline before. After moving things out to use usecallback, it's not needed anymore. removed!